tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/RetryConditionTest.java

```diff
@@ -215,4 +227,144 @@ public void testClassLevelRetryWithAbortOnTrue() {
         Assert.assertEquals(clientForClassLevelAbortOn.getRetryCountForWritingService(), 1,
             "The max invocation counter should be 1 as the abort condition is true");
     }
+
+
+    /**
+     * Persistent Error condition. Will retry 2 times and still throw exception.
+     * ServiceA uses {@link org.eclipse.microprofile.faulttolerance.Asynchronous} and will always return IOException.
+     */
+    @Test
+    public void testAsyncRetryExceptionally() {
+        final CompletionStage<String> future = asyncRetryClient.serviceA();
+
+        assertCompleteExceptionally(future, IOException.class, "Simulated error");
+        assertEquals(2, asyncRetryClient.getCountInvocationsServA());
+    }
+
+    /**
+     * Persistent Error condition inside a CompletableFuture.
+     * Will not retry because method is not marked with @Asynchronous
+     * ServiceB will always complete exceptionally with IOException.
+     */
+    @Test
+    public void testNoAsynWilNotRetryExceptionally() {
+        CompletableFuture<String> future = new CompletableFuture<>();
+
+        asyncRetryClient.serviceBFailExceptionally(future);
+
+        assertCompleteExceptionally(future, IOException.class, "Simulated error");
+        // no retries
+        assertEquals("No retries are expected",0, asyncRetryClient.getCountInvocationsServBFailExceptionally());
+    }
+
+    /**
+     * Persistent Error condition outside the CompletableFuture.
+     * Will retry because
+     * ServiceB will always throw IOException.
+     */
+    @Test
+    public void testNoAsynRetryOnMethodException() {
+        CompletableFuture<String> future = new CompletableFuture<>();
+
+        try {
+            asyncRetryClient.serviceBFailException(future);
+            fail("Was expecting an exception");
+        }
+        catch (RuntimeException e) {
+            assertEquals("Simulated error", e.getMessage());
+        }
+        // 2 retries
+        assertEquals(2, asyncRetryClient.getCountInvocationsServBFailException());
+    }
+
+    /**
+     * Temporary error. Will retry 3 times, the first 2 will fail.
+     * ServiceC uses {@link org.eclipse.microprofile.faulttolerance.Asynchronous}.
+     */
+    @Test
+    public void testRetrySuccess() {
+        final CompletionStage<String> future = asyncRetryClient.serviceC();
+
+        assertCompleteOk(future, "Success");
+        assertEquals(2, asyncRetryClient.getCountInvocationsServC());
```

Comment:
Looking at `serviceC` I think we expect 3 invocations here: 2 failures and 1 success